public/static/app/perf-bb/PerfBBRejectsCtrl.js

```diff
@@ -0,0 +1,385 @@
+mciModule.controller('PerfBBRejectsCtrl', function (
+  $scope, $window, EvgUiGridUtil, EvgUtil, FORMAT, MDBQueryAdaptor, uiGridConstants,
+  STITCH_CONFIG, Stitch, Settings, $timeout, $compile, $log, WhitelistService, $q, RejectState
+) {
+  // Perf Rejects View-Model.
+  const vm = this;
+  const project = window.project;
+  const LIMIT = 2000;
+
+  const updateToWhitelist = function(items, func, mark) {
+    const task_revisions = _.chain(items).
+                              map((item) => _.pick(item, 'revision', 'project', 'variant', 'task')).
+                              uniq((item) => _.values(item).join('-')).
+                              value();
+    $log.debug(task_revisions);
+    const promise = func(task_revisions);
+    promise.then(function(ok) {
+      if (!ok) return;
+      _.each(task_revisions, (task_revision) => {
+        _.chain(vm.gridOptions.data).where(task_revision).each((doc)=> {
+          doc.whitelisted = mark;
+        }).value();
+      });
+      refreshGridData();
+      // Call vm.reload() to do a full reload from the server.
+    });
+
+  };
+
+  $scope.addWhitelist = function(items) {
+    return updateToWhitelist(items, WhitelistService.addWhitelist, "✔");
+  };
+
+  $scope.removeWhitelist = function(items) {
+    return updateToWhitelist(items, WhitelistService.removeWhitelist);
+  };
+
+  function refreshGridData() {
+    vm.gridApi.selection.clearSelectedRows();
+    handleRowSelectionChange(vm.gridApi);
+  }
+
+  // Holds currently selected items.
+  vm.selection = [];
+  vm.actions = [
+    {
+      title: 'Add',
+      action: $scope.addWhitelist,
+      disabled: () => vm.selection.length == 0 || _.all(vm.selection, (doc)=> doc.whitelisted)
+    },
+    {
+      title: 'Remove',
+      action: $scope.removeWhitelist,
+      disabled: () => vm.selection.length == 0 || _.all(vm.selection, (doc)=> !doc.whitelisted)
+    },
+  ];
+
+  vm.state = new RejectState(project, vm, $scope);
+
+  // Creates aggregation expression, which could be used by Stitch
+  // for given `state`
+  $scope.getAggChain = (state) => {
+    let chain = [];
+
+    // Check if the state has filters
+    let filteringChain = MDBQueryAdaptor.compileFiltering(
+      // filters context enhances state data with important meta data
+      state.getFilteringContext()
+    );
+
+    if(!filteringChain) {
```

Comment:
[nit] Everywhere else in this file, there is a space between the 'if' and the '(', could we do that here and on the next 'if' as well?

Reply:
:+1: